nitro-protocol/contracts/MultiAssetHolder.sol

```diff
@@ -57,10 +57,18 @@ contract MultiAssetHolder is IMultiAssetHolder, StatusManager {
             require(msg.value == amount, 'Incorrect msg.value for deposit');
         } else {
             // require successful deposit before updating holdings (protect against reentrancy)
-            require(
-                IERC20(asset).transferFrom(msg.sender, address(this), amountDeposited),
-                'Could not deposit ERC20s'
+            uint256 before = IERC20(asset).balanceOf(address(this));
+            (bool success, ) = asset.call(
+                abi.encodeWithSignature(
+                    'transferFrom(address,address,uint256)',
+                    msg.sender,
+                    address(this),
+                    amountDeposited
+                )
             );
+            require(success, 'Could not deposit ERC20s');
+            uint256 aft = IERC20(asset).balanceOf(address(this));
+            require(aft == before + amountDeposited, 'Could not deposit ERC20s');
```

Comment:
"Bad" ERC20 tokens do not return anything on successful deposit. We could simply rely on them reverting if the depositor has insufficient tokens (instead of using `balanceOf` to do a check, before and after the deposit)?